### DISCUSSION TITLE
"Recently Added" sort functionality for songs

### DISCUSSION THREAD

**Discussion by puresick** [navidrome/navidrome#2847] [Ideas]
Having a sort function for songs to order based on when they are added would help in certain situations where single audio files are regularly been added to the library.

In my setup, I am fetching music podcasts regularly and add them to my library. Each podcast is its own album with the episodes being tracks of that album. If an album gets new tracks it does not move in the "Recently Added" section, which IMO is the correct behaviour.
Unfortunately that makes it cumbersome to check if new podcast episodes have been added to the library.

Having a "Recently Added" section for songs would make this a no-brainer: Just check that sorting and see whats on top of the list.

What do you think, is this a feature worth adding?

**Comment by deluan** [comment]
This is already possible in the current UI:


https://github.com/navidrome/navidrome/assets/331353/f7b71b4c-795e-4664-a92c-d00a4180476a


Is this what you are looking for?

**Reply by puresick** [reply]
I.. ehm.. :facepalm: 

First of all let me excuse for wasting your time. This is indeed what I was looking for. :+1: Thank you!

I apparently never realized that I can scroll this popover since no indicator for that interaction is shown and modern operating systems tend to hide the scrollbar by default.

It might be worth exploring how the scrollable area could be made more obvious. If you're open for it I am happy to offer my help for working on this (I am a frontend developer myself — which makes that whole situation even more embarrasing :sweat_smile:).

What do you think?

**Reply by deluan** [reply]
No worries, you are not the first :)

We are not changing this in this version. In the new one, that we rewriting to use React-Admin 4, we will use this component: https://marmelab.com/react-admin/SelectColumnsButton.html Not sure how it feels when it has more fields than the view.

**Reply by puresick** [reply]
Phew! :)

Alright, noted!
Once you implement this in the new version, feel free to hit me up if you need help testing it. :)

Thank you again! :pray:

**Comment by peef2** [comment]
<img width="262" height="620" alt="image" src="https://github.com/user-attachments/assets/efef95c6-0218-49b9-971e-d9f9571c00c4" />

hello, sorry for reviving this, but i noticed the sort date gets a little weird with AM and PM. in screenshot, i am sorting to most recent, but the 12 pm ones end up after the ones that were from 2-10 pm.
